api/handler.js

```diff
@@ -15,18 +21,31 @@ module.exports.courses = async event => {
     url += `&start_rank=${offset}`;
   }
 
-  const response = await fetch(url, {
-    method: "GET",
-    headers: {
-      'Content-Type': 'application/json'
-    }
-  });
+  try {
+    const response = await fetch(url, {
+      method: "GET",
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    });
 
-  const json = await response.json();
+    checkResponse(response);
 
-  return {
-    statusCode: 200,
-    body: JSON.stringify(json),
-  };
+    const body = await response.json();
+
+    result.body = JSON.stringify(body);
```

Comment:
It seems a little odd to me to parse JSON into JS and then stringify it back to JSON. Maybe that's necessary. Would the following work instead?
```suggestion
    result.body = await response;
```

Reply:
I see where you're coming from, but I don't know if this is possible. I don't particularly want to parse the JSON into an object, but the `.json()` method is the only way I know of getting the json body from the response. `response` and `response.body` are much more complex objects than just the body/content of the response.

Reply:
Fair enough. Also, I think at some point soon we will need to make a few amendments to the data that comes from Funnelback before we return it (Funnelback data contains some `"true"`/`"false"` strings that need to be converted to booleans to adhere to the Courses API specification), so it will be necessary to have the Funnelback response as a JS object to do that, before then stringifying it to JSON for returning.